src/lib/Microsoft.Health.Extensions.Fhir/MultipleResourceFoundException.cs

```diff
@@ -43,14 +44,14 @@ public MultipleResourceFoundException(string message, Exception innerException)
 
         public override string Operation => ConnectorOperation.FHIRConversion;
 
-        private static string GenerateErrorMessage<TResource>(int resourceCount, IEnumerable<string> ids)
+        private static string GenerateErrorMessage<TResource>(int resourceCount, IEnumerable<IResourceMetadata> metadata)
         {
             var sb = new StringBuilder($"Multiple resources {resourceCount} of type {typeof(T)} found, expected one.");
 
-            if (ids != null)
+            if (metadata != null)
             {
-                sb.Append(" Resource internal ids: ");
-                sb.Append(string.Join(", ", ids));
+                sb.Append(" Resource metadata: ");

```

Comment:
Nit; the Append function returns a Stringbuilder. So you should be able to chain these calls together.